### PR TITLE
Fix MET Significance, Covariance Matrix and unclustered energy lepton subtraction [10_6_X]

### PR DESCRIPTION
**Pull request by ahinzmann**
#### PR description:

Resolve issue #25849

selectedPatLeptons have no sourceCandidatePtrs linked to them.
Therefore lepton subtraction from unclustered energy fails in METSignficance.cc.
slimmedLeptons have properly linked sourceCandidatePtrs and allows lepton subtraction from unclustered energy.
This resulted in a disagreement between MET significance, unclustered energy and covariance matrix stored in MiniAOD and recomputed out of MiniAOD.
The value in MiniAOD was wrong and is now fixed by using slimmedLeptons as well.

#### PR validation:

It was checked that METSignficance.cc now sees all the lepton PF candidates, which were previously not seen on workflow 136.8311.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Backport of #29385.

### CODE REVIEW

**Comment by ahinzmann** [comment]
(no need to test this one until #29385 is integrated)

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
backport of #29385

File: PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py [hunk 1]
```diff
@@ -1426,19 +1480,26 @@ def recomputeRawMetFromPfcs(self, process, pfCandCollection, onMiniAOD, patMetMo
...
+
+                if self.getvalue("Puppi"):
+                    getattr(process, _myPatMet).srcPFCands = cms.InputTag('puppiForMET')
+                    getattr(process, _myPatMet).srcJets = cms.InputTag('cleanedPatJets'+postfix)
+                    getattr(process, _myPatMet).srcJetSF = cms.string('AK4PFPuppi')
+                    getattr(process, _myPatMet).srcJetResPt = cms.string('AK4PFPuppi_pt')
+                    getattr(process, _myPatMet).srcJetResPhi = cms.string('AK4PFPuppi_phi')
 
 
     def extractMET(self, process, correctionLevel, patMetModuleSequence, postfix):
```
Comment (slava77) [inline PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py:1505]: below, in ```def miniAODConfiguration(``` in #29385 there is a removal of one modification
```diff
-                getattr(process, "patPFMet"+postfix).srcLeptons = -cms.VInputTag(self._parameters["electronCollection"].value, 
-                                                                                self._parameters["muonCollection"].value,
-                                                                                self._parameters["photonCollection"].value,
-                                                                                )
```

was it intentionally left out here? .. or was it not supposed to be in the master?

Comment (ahinzmann) [reply PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py:1505]: this modification is effectively overwritten by the other setting of srcLeptons added in this PR, thus it is dropped in the master. However, here it is kept for the case where run2_miniAOD_devel is disabled.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/29392	dc229c9 
- code changes are in line with the PR description and the master version PR. Differences in the implementation, compared with the master are related to preserving the default behavior; the new behavior is enabled with ```run2_miniAOD_devel```
- jenkins tests pass and comparisons with the baseline show

**Comment by silviodonato** [comment]
do you have comments @santocch?

**Comment by santocch** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**